apps/desktop/desktop_native/core/src/ssh_agent/mod.rs

```diff
@@ -0,0 +1,136 @@
+use std::sync::Arc;
+
+use anyhow::Error;
+#[cfg(windows)]
+use async_stream::stream;
+#[cfg(windows)]
+use futures::stream::{Stream, StreamExt};
+
+use ssh_agent::Key;
+use std::collections::HashMap;
+use std::sync::RwLock;
+use tokio::sync::Mutex;
+
+pub mod msg;
+pub mod ssh_agent;
+pub mod russh_encoding;
+#[cfg(windows)]
+pub mod namedpipelistenerstream;
+
+#[cfg(unix)]
+use tokio::net::UnixListener;
+#[cfg(unix)]
+use homedir::my_home;
+
+static KEYSTORE: std::sync::LazyLock<ssh_agent::KeyStore> =
+    std::sync::LazyLock::new(|| ssh_agent::KeyStore(Arc::new(RwLock::new(HashMap::new()))));
+
+#[derive(Clone)]
+struct SecureAgent {
+    send_tx: tokio::sync::mpsc::Sender<String>,
+    response_rx: Arc<Mutex<tokio::sync::mpsc::Receiver<bool>>>,
+}
+
+impl ssh_agent::Agent for SecureAgent {
+    async fn confirm(&self, ssh_key: Key) -> bool {
+        // make sure we will recv our response by locking the channel
+        let mut rx_channel = self.response_rx.lock().await;
+        self.send_tx.send(ssh_key.cipher_uuid.clone()).await.unwrap();
```

Comment:
Nit: I don't think this clone is needed here, we're taking the key by value:
```suggestion
        self.send_tx.send(ssh_key.cipher_uuid).await.unwrap();
```